src/AddAttributesTwigExtension.php

```diff
@@ -1,36 +1,35 @@
 <?php
 
 namespace Drupal\emulsify_twig;
+
 use Drupal\Core\Template\Attribute;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
 
 /**
- * Class DefaultService
+ * Class DefaultService.
  *
  * @package Drupal\EmulsifyExt
  */
-class AddAttributesTwigExtension extends \Twig_Extension {
-  /**
-   * {@inheritdoc}
-   * This function must return the name of the extension. It must be unique.
-   */
-  public function getName() {
-    return 'emulsify_twig_add_attributes';
-  }
+class AddAttributesTwigExtension extends AbstractExtension {
 
   /**
    * In this function we can declare the extension function.
    */
   public function getFunctions() {
-    return array(
-      new \Twig_SimpleFunction('add_attributes', array($this, 'add_attributes'), array('needs_context' => true, 'is_safe' => array('html'))),
-    );
+    return [
+      new TwigFunction(
+        'add_attributes',
+        [$this, 'addAttributes'],
+        ['needs_context' => TRUE, 'is_safe' => ['html']]
+      ),
+    ];
   }
 
-  /*
-   * This function is used to return alt of an image
-   * Set image title as alt.
+  /**
+   * This function is used to return alt of an image. Set image title as alt.
    */
-  public function add_attributes($context, $additional_attributes = []) {
+  public function addAttributes($context, $additional_attributes = []) {
```

Comment:
@marcb4k I would consider this a breaking change.  In order to get Semantic Release to bump the major version you need the text `BREAKING CHANGE:` (and some text description, don't forget the colon) in the commit message.  You could create an empty commit with this in the description if you have no code changes to make and semantic release will pick it up.

Reply:
The twig function called earlier on line 23 simply points to this new internal name. No change required for the actual twig function name (`add_attributes`), so I don't think this is a breaking change specifically, however, your comments about the 'AbstractExtension` are definitely worth looking into and I'll update the `core_version_requirement` after I research when that new class was added to Drupal. Thanks!